torch/_inductor/decomposition.py

```diff
@@ -261,15 +265,11 @@ def bmm(
     batch2: torch.Tensor,
 ) -> torch.Tensor:
     if config.coordinate_descent_tuning and self.device.type != "cpu":
-        if guard_size_oblivious(self.shape[1] == 1) or guard_size_oblivious(
-            batch2.shape[2] == 1
-        ):
+        if guard_or_false(self.shape[1] == 1) or guard_or_false(batch2.shape[2] == 1):
```

Comment:
I've been thinking about this line (and examples like it) and what behavior we ACTUALLY want. Do we REALLY want something like:
```
if statically_known_true(self.shape[1] == 1 or batch2.shape[2] == 1):
    guard_or_false(self.shape[1] == 1) or guard_or_false(batch2.shape[2] == 1)
```
In other words - if we can determine this using the currently known information then do so and guard on whatever we needed otherwise don't add ANY guards.

Reply:
if we already know the value statically , then there is no value to guard right? guard wont guard in fact.